text_test.go

```diff
@@ -0,0 +1,53 @@
+package pflag
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+)
+
+func setUpTime(t *time.Time) *FlagSet {
+	f := NewFlagSet("test", ContinueOnError)
+	f.TextVar(t, "time", time.Now(), "time stamp")
+	return f
+}
+
+func TestText(t *testing.T) {
+	testCases := []struct {
+		input    string
+		success  bool
+		expected time.Time
+	}{
+		{"2003-01-02T15:04:05Z", true, time.Date(2003, 1, 2, 15, 04, 05, 0, time.UTC)},
+		{"2003-01-02 15:05:01", false, time.Date(2002, 1, 2, 15, 05, 05, 07, time.UTC)},
```

Comment:
This test case is confusing me: it's providing a string which I assume will fail parsing (?) but then it also provides a `time.Date` instance that doesn't match. That makes it hard to see, at a glance, _why_ we expect this test to fail.

Can we maybe provide an empty `time.Date` instance to signal that we don't actually expect to look at it in the test?

```suggestion
		{"2003-01-02 15:05:01", false, time.Time{}}, // invalid date string
```